src/routes/clients.js

```diff
@@ -61,14 +71,26 @@ router.post('/', (req, res) => {
       case 'name':
         errno = ERRNO_INVALID_API_CLIENT_NAME;
         break;
+      case 'permissions':
+        errno = ERRNO_INVALID_API_CLIENT_PERMISSION;
+        break;
       default:
         errno = ERRNO_BAD_REQUEST;
     }
     return ApiError(res, 400, errno, BAD_REQUEST);
   }
 
   db().then(models => {
-    models.Clients.create(req.body).then(client => {
+    let client;
+    models.sequelize.transaction(transaction => {
```

Comment:
Hmm... that's not how I understood the [doc](http://docs.sequelizejs.com/en/latest/docs/transactions/#managed-transaction-auto-callback). IIUC line 94 is not executed until the Promise returned at line 86 resolves.